app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -75,6 +76,20 @@ def validate_authorize_form
       end
     end
 
+    def force_login_if_prompt_param_is_login_and_request_is_external
+      return unless user_signed_in? && @authorize_form.prompt == 'login'
+      sign_out unless referring_host.to_s.casecmp(request.host).zero?
```

Comment:
What do you think about comparing with `Figaro.env.domain_name` instead of `request.host`? That way, it eliminates the chances that the value of `request.host` is not what we expect, such as if someone spoofs the `HTTP_HOST` header.

Reply:
Also, can we write a test for this please?

Reply:
If someone spoofs the referrer or host the worst they can wind up doing here is resetting their own session.  However, I think that depending on the env setting is a great idea and I will implement it.